tutorial/08-diffusion.ipynb

```diff
@@ -123,22 +150,68 @@
     "        left, l = lpoint(idx, n, coords, shape, m)\n",
     "        right, r = rpoint(idx, n, coords, shape, m)\n",
     "        c = coords[n]\n",
-    "        lptag += (((tag[right] - tag[ent])/(r-c)) - ((tag[ent] - tag[left])/(c-l))) / ((r-l)/2)\n",
+    "\n",
+    "        # Retrieve the tag values for the entities\n",
+    "        tag_data_ent = m.mesh.tag_get_data(tag, ent)[0]\n",
```

Comment:
```python
Cell In[1], line 47, in laplace(tag, idx, m, shape)
     45     right, r = rpoint(idx, n, coords, shape, m)
     46     c = coords[n]
---> 47     lptag += (((tag[right] - tag[ent]) / (r - c)) - ((tag[ent] - tag[left]) / (c - l))) / ((r - l) / 2)
     48 return lptag

TypeError: 'pymoab.tag.Tag' object is not subscriptable
```
I am using `MOAB` 5.4 and also checked 5.3. We can't use the subscript operator `[]` on a `pymoab.tag.Tag` object. The [Core](https://bitbucket.org/fathomteam/moab/src/902f803a01dbe8e279c96e54c7106e9da7139875/pymoab/pymoab/core.pyx#lines-886) is updated.

Reply:
I think we need to rethink this whole file updated with the newer (updated by @pshriwise 4 years ago...) mesh infrastructure.

Reply:
it appears that this file was updated with a couple of small changes as the interface changed but was never tested.  I think it may be best to remove and/or rewrite it from scratch.

In particular, I am a little lost on how to best get the coordinates correctly with the current interface.  The last time this code worked, we got a single set of coordinates from a passing in a structured mesh hex (which has 8 vertices) - and I'm not sure how that was behaving.

Reply:
While considering rewriting from scratch, presents an opportunity to integrate new built-in methods, simplifying tasks like timesteps, rendering, and reactor management.